### ISSUE TITLE
OrderGridCollectionFilter Plugin Breaks Mapping for created_at Column

### ISSUE THREAD

**Issue by sgeleon** [magento/magento2#38916]
### Preconditions and environment

- Magento v2.4.7-p1

### Steps to reproduce

STEPS TO REPRODUCE
Sales > Orders
Filters > set Purchase Date from and to
Click Apply Filters

![image](https://github.com/magento/magento2/assets/18437080/f21f826c-09ff-4a58-97c7-50088c1dd6fc)

### Expected result

The filters are applied.

### Actual result

'Something went wrong with processing the default view and we have restored the filter to its original state.

### Additional information

This problem is same like from [this bug report.](https://github.com/magento/magento2/issues/36439)

This plugin is fixed it.

```
Index: vendor/magento/module-sales/Plugin/Model/ResourceModel/Order/OrderGridCollectionFilter.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/vendor/magento/module-sales/Plugin/Model/ResourceModel/Order/OrderGridCollectionFilter.php b/vendor/magento/module-sales/Plugin/Model/ResourceModel/Order/OrderGridCollectionFilter.php
--- a/vendor/magento/module-sales/Plugin/Model/ResourceModel/Order/OrderGridCollectionFilter.php	
+++ b/vendor/magento/module-sales/Plugin/Model/ResourceModel/Order/OrderGridCollectionFilter.php	(date 1717557636000)
@@ -56,7 +56,7 @@
             }
 
             $fieldName = $subject->getConnection()->quoteIdentifier($field);
-            $condition = $subject->getConnection()->prepareSqlCondition($fieldName, $condition);
+            $condition = $subject->getConnection()->prepareSqlCondition('`main_table`.'.$fieldName, $condition);
             $subject->getSelect()->where($condition, null, Select::TYPE_CONDITION);
 
             return $subject;
```

### Triage and priority

- [X] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Dash** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Dash** [comment]
Hi @sgeleon 

Thanks for reporting and collaboration.

Verified the issue on Magento 2.4 dev instance but the issue is not reproducible.

However, we can reproduce the issue on 2.4.7-p1.

The issue might have been fixed in the latest 2.4 dev branch.

Steps to reproduce
STEPS TO REPRODUCE
Sales > Orders
Filters > set Purchase Date from and to
Click Apply Filters

Please refer the attached screen-recording.

In 2.4 dev instance:

https://github.com/user-attachments/assets/00da1345-83b3-47ed-9afa-85e29b0e50cf

In 2.4.7-p1:

<img width="1440" alt="Screenshot 2024-07-16 at 4 22 04 PM" src="https://github.com/user-attachments/assets/1eb14630-3d12-46ed-91a8-9b71eaf810a9">

**Comment by hclauss-ltk** [comment]
Hello together, 

we have the same issue and can´t filter our products anymore, same error applies. Also since the error occurs, we can´t even see the unfiltered product overview anymore. Are there any updates how to reset this? 

Thanks in advance!

**Comment by engcom-Dash** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Dash** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Dash** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Dash** [comment]
Hi @sgeleon 

We have verified the issue on 2.4 dev instance but the issue is not reproducible.

Please look into the screenshots attached.

<img width="1440" alt="Screenshot 2024-07-23 at 11 04 06 AM" src="https://github.com/user-attachments/assets/83220323-3cb8-44b2-b052-2989fb0af059">

Let us know if we missed anything.

Thanks.

**Comment by Bashev** [comment]
Here is the fix, but it's never released. https://github.com/magento/magento2/blame/11e7d89380c3ef9b9bd18c1d8b7815b0fad96e34/app/code/Magento/Sales/Plugin/Model/ResourceModel/Order/OrderGridCollectionFilter.php#L58C75-L58C75

@engcom-Dash can you confirm, this will comes with next release?

**Comment by Bashev** [comment]
@engcom-Hotel what kind of update is required here? As i mentioned above, PR is merged (https://github.com/magento/magento2/commit/f09c22200da0e9cb4481c953ebc0d1ce3a226f38), but somehow is not released in 2.4.7-p2

**Comment by engcom-Hotel** [comment]
Hello @Bashev,

Thanks for your query!

We have started a discussion internally on your query and will reply you back ASAP on this. 

Meanwhile moving this issue on hold.

Thanks

**Comment by engcom-Hotel** [comment]
Hello @Bashev,

We have just received a reply from the internal team. The fix will be released in the 2.4.8 version.

Thanks

**Comment by dchaykas** [comment]
@engcom-Hotel will the fix address https://github.com/magento/magento2/issues/39000 as well?
